src/java/org/apache/cassandra/metrics/InternodeInboundMetrics.java

```diff
@@ -42,12 +46,41 @@ public class InternodeInboundMetrics
     private final MetricName throttledCount;
     private final MetricName throttledNanos;
 
+    /**
+     * Factory method to create metrics for given inbound message handlers.
+     * This ensures the metrics provider is used when configured.
+     *
+     * @param peer IP address and port to use for metrics label
+     * @param handlers the inbound message handlers
+     * @return new InternodeInboundMetrics instance
+     */
+    public static InternodeInboundMetrics create(InetAddressAndPort peer, InboundMessageHandlers handlers)
+    {
+        if (CUSTOM_INTERNODE_INBOUND_METRICS_PROVIDER_PROPERTY.isPresent())
+        {
+            Class<InternodeInboundMetrics> klass = FBUtilities.classForName(CUSTOM_INTERNODE_INBOUND_METRICS_PROVIDER_PROPERTY.getString(), "Internode Inbound Metrics Provider");
+            try
+            {
+                return klass.getDeclaredConstructor(InetAddressAndPort.class, InboundMessageHandlers.class).newInstance(peer, handlers);
+            }
+            catch (NoSuchMethodException | InstantiationException | IllegalAccessException |
+                   InvocationTargetException e)
+            {
+                throw new RuntimeException(e);
```

Comment:
what happens to this ? 

do we want that consequence, or to log the error and fallback to the `InternodeOutboundMetrics` default ?